torch/ao/quantization/_quantize_pt2e.py

```diff
@@ -0,0 +1,205 @@
+        conv_args = list(conv_op.args)
+        # calling data since the fused_weight and fused_bias are nn.Parameter
+        weight_attr_name = conv_args[1].target
+        setattr(m, weight_attr_name, fused_weight)
+        if conv_args[2] is not None:
+            bias_attr_name = conv_args[2].target
+        else:
+            bias_attr_name = weight_attr_name + "_bias"
+            with m.graph.inserting_before(conv_op):
+                get_bias_node = m.graph.get_attr(bias_attr_name)
+            conv_args[2] = get_bias_node
+        setattr(m, bias_attr_name, fused_bias)
+        conv_op.args = tuple(conv_args)
+
+        # native_batch_norm has 3 outputs, we expect getitem calls on the output
+        # and we want to replace the uses of getitem 0 with the output of conv
+        #
+        # Before:
+        # conv -> bn - (first output) -> users1
+        #          \ - (second output) -> users2
+        #          \ - (third output) -> users3
+        # After:
+        # conv -> (first output) -> users1
+        #       bn -
+        #          \ - (second output) -> users2
+        #          \ - (third output) -> users3
+        # if users2 and users3 are empty then bn will be removed through dead code elimination
+
+        for user in bn_op.users:
+            if user.op != "call_function" or user.target != operator.getitem or user.args[1] != 0:
+                continue
+            user.replace_all_uses_with(conv_op)
+    m.graph.eliminate_dead_code()
+    m.recompile()
+
+def _rearrange_weight_observer_for_addmm(
+    model: GraphModule,
+) -> None:
+    """
+    before:
+         weight - t - observer \
+          input - observer - addmm
+    after:
+         weight - observer - t \
+           input - observer - addmm
+    """
+    named_modules = dict(model.named_modules(remove_duplicate=False))
+    for node in model.graph.nodes:
+        if node.target != torch.ops.aten.addmm.default:
+            continue
+        addmm = node
+        maybe_weight_obs = addmm.args[2]
+        if not is_activation_post_process_node(maybe_weight_obs, named_modules):
+            continue
+        transpose_node = maybe_weight_obs.args[0]
+        if transpose_node.target != torch.ops.aten.t.default:
+            continue
+        # swap the order of transpose and observation
+
+        maybe_weight_obs.replace_input_with(transpose_node, transpose_node.args[0])
+        # remove the transpose node
+        with model.graph.inserting_after(maybe_weight_obs):
+            args = list(transpose_node.args)
+            args[0] = maybe_weight_obs
+            new_transpose_node = model.graph.create_node(
+                "call_function",
+                torch.ops.aten.t.default,
+                tuple(args),
+                transpose_node.kwargs
+            )
+        addmm.replace_input_with(maybe_weight_obs, new_transpose_node)
+
+    model.graph.eliminate_dead_code()
+    model.graph.lint()
+
+def prepare_pt2e(
```

Comment:
can this function do a check that `model` has to be an output of torchdynamo export, and it's not expected to work on any GraphModule?

Reply:
sounds good, not sure how we can do this, @qihqi @SherlockNoMad is there a way to check this now?

Reply:
probably not essential now, ok to punt IMO